sklearn/linear_model/least_angle.py

```diff
@@ -1437,7 +1438,7 @@ def fit(self, X, y, copy_X=True):
 
         self.alphas_ = alphas_
         with np.errstate(divide='ignore'):
-            self.criterion_ = n_samples * np.log(mean_squared_error) + K * df
+            self.criterion_ = n_samples * mean_squared_error/sigma + K * df
```

Comment:
to me it should be 

mean_squared_error / sigma2 + K * df

given that K is 2 or log(n)

ideally everything should be divided by n.

Did you check how it affects the example using BIC and AIC?